File: apps/old-api/src/main/java/pt/iade/ei/thinktoilet/models/response/ApiResponse.java
package pt.iade.ei.thinktoilet.models.response;

import lombok.Data;

import java.time.LocalDateTime;

@Data
public class ApiResponse {
    private int status;
    private String message;
    private String timestamp;

    public ApiResponse(int status, String message) {
        this.status = status;
        this.message = message;
        this.timestamp = LocalDateTime.now().toString();
    }
}

File: app/app/src/main/java/pt/iade/ei/thinktoilet/models/responses/ApiResponse.kt
package pt.iade.ei.thinktoilet.models.responses

import com.google.gson.annotations.SerializedName

data class ApiResponse(
    @SerializedName("status") val status: Int,
    @SerializedName("message") val message: String,
    @SerializedName("timestamp") val timestamp: String
)
